include/onnxruntime/core/session/onnxruntime_c_api.h

```diff
@@ -3329,6 +3331,38 @@ struct OrtApi {
   ORT_API2_STATUS(AddExternalInitializers, _In_ OrtSessionOptions* options,
                   _In_reads_(input_len) const char* const* initializer_names,
                   _In_reads_(input_len) const OrtValue* const* initializers, size_t initializers_num);
+
+  ORT_API2_STATUS(CreateAttribute,
+                  _In_ const char* name,
+                  _In_ const void* data,
+                  _In_ int len,
+                  _In_ ONNXTensorElementDataType type,
+                  _In_opt_ int is_array,
+                  _Outptr_ OrtOpAttr* op_attr);
+
+  ORT_API2_STATUS(ReleaseAttribute, _Inout_ OrtOpAttr* op_attr);
+
+  ORT_API2_STATUS(CreateOperator,
+                  _In_ const OrtKernelInfo* info,
+                  _In_ const char* op_name,
+                  _In_ const char* domain,
+                  _In_ int version,
+                  _In_opt_ const char** type_constraint_names,
+                  _In_opt_ const ONNXTensorElementDataType* type_constraint_values,
+                  _In_opt_ int type_constraint_count,
+                  _In_opt_ const OrtOpAttr* attr_values,
+                  _In_opt_ int attr_count,
+                  _Outptr_ OrtOp* ort_op);
+
+  ORT_API2_STATUS(InvokeOperator,
+                  _In_ const OrtKernelContext* context,
+                  _In_ const OrtOp ort_op,
+                  _In_ const OrtValue* const* input_values,
```

Comment:
This API assumes that users will pass the input values in the same order as specified in the ONNX spec. Is that correct? How do we validate this? And where do we document this?

Reply:
Added comment for all new APIs - for InvokeOp, added words state that "must follow the order as onnx specification".
Will doc the feature thoroughly in the gh-pages branch.